luarules/gadgets.lua

```diff
@@ -1595,10 +1611,21 @@ function gadgetHandler:UnitPreDamaged(unitID, unitDefID, unitTeam, damage, paral
 			attackerID, attackerDefID, attackerTeam)
 
 		if dmg ~= nil then
+			if baseDamage ~= retDamage and UnitPreDamaged_damage_caution_count == 0 then
+				-- to caution devs if multiple gadgets are trying to adjust damage
+				Spring.Echo("GadgetHandler: caution, multiple gadgets influencing damage")
+				UnitPreDamaged_damage_caution_count = UnitPreDamaged_damage_caution_count + 1
+			end
 			retDamage = dmg
 		end
 		if imp ~= nil then
-			retImpulse = imp
+			if retImpulse ~= 1.0 and imp ~= 1 and UnitPreDamaged_impulse_caution_count == 0 then
+			    -- to caution devs if multiple gadgets are trying to adjust impulse
+				Spring.Echo("GadgetHandler: caution, multiple gadgets influencing impulse")
+				UnitPreDamaged_impulse_caution_count = UnitPreDamaged_impulse_caution_count + 1
+			end
+			--assuming multiple gadgets influencing impulse should stack multiplicative
+			retImpulse = retImpulse*imp
```

Comment:
Personally I'd keep just this single line out of this whole PR (but it's an excellent change).